# Práctica 1.1 — Probabilidad y Estadística

## Aarón Martínez Garcidueñas - 511

### Análisis de la evolución poblacional mundial

### Objetivo

Utilizar un conjunto de datos real para comprobar el funcionamiento del entorno de análisis de datos y aplicar medidas estadísticas descriptivas básicas utilizando Pandas y NumPy.

## Actividad 1 — Identificación de la fuente

### Cita de la fuente:

["Annual change in population"](https://ourworldindata.org/grapher/annual-population-growth?v=1&csvType=full&useColumnShortNames=false) on the Our World in Data website.

United Nations, World Population Prospects (2024)

UN, World Population Prospects (2024) – processed by Our World in Data. “Annual change in population – UN WPP” [dataset]. United Nations, “World Population Prospects”; United Nations, “World Population Prospects - Interim Update” [original data].

1.	**¿Quién produjo originalmente los datos?**
Las naciones unidas en World Population Prospects (2024)
2.	**¿Qué organización los procesa para su utilización en Our World in Data?**
Los procesa la organización Our World in Data para su sitio
3.	**¿Qué representa la variable que analizarás?**
El cambio de la población mundial y sus proyecciones

## Actividad 2 - Carga y exploración de datos

In [2]:
import pandas as pd
import numpy as np

In [3]:
df = pd.read_csv("../data/raw/annual-population-growth/annual-population-growth.csv")

In [4]:
df.head()

,Entity,Code,Year,Annual change in population,Annual population change (Projected)
0,Afghanistan,AFG,1951,103163.0,NaN
1,Afghanistan,AFG,1952,108441.0,NaN
2,Afghanistan,AFG,1953,108919.0,NaN
3,Afghanistan,AFG,1954,111251.0,NaN
4,Afghanistan,AFG,1955,119027.0,NaN


In [5]:
df.shape

(38400, 5)

In [6]:
df.columns.tolist()

['Entity',
 'Code',
 'Year',
 'Annual change in population',
 'Annual population change (Projected)']

In [7]:
df.dtypes

Entity                                      str
Code                                        str
Year                                      int64
Annual change in population             float64
Annual population change (Projected)    float64
dtype: object

In [8]:
df.describe()

,Year,Annual change in population,Annual population change (Projected)
count,38400.000000,1.868800e+04,1.971200e+04
mean,2025.500000,2.063019e+06,8.375481e+05
std,43.300872,9.440859e+06,5.816745e+06
min,1951.000000,-3.315935e+06,-2.291035e+07
25%,1988.000000,1.781000e+03,-9.742750e+03
50%,2025.500000,4.689800e+04,9.350000e+01
75%,2063.000000,3.021128e+05,1.411160e+05
max,2100.000000,9.337137e+07,7.206736e+07


In [9]:
df.isnull().sum()

Entity                                      0
Code                                     1200
Year                                        0
Annual change in population             19712
Annual population change (Projected)    18688
dtype: int64

In [10]:
print(df["Entity"].nunique())

256


1.	**¿Cuántas filas y columnas tiene el dataset?**
38,400 filas y 5 columnas
2.	**¿Qué tipo de datos contiene la variable Year?**
Contiene datos del tipo entero (int)
3.	**¿Existen valores faltantes?**
Si, de código y de cambio anual en población normal y proyectado.
4.	**¿Cuántas entidades diferentes contiene el dataset?**
256


## Actividad 3 - Estadística descriptiva

In [11]:
df_extration_filas = df[df["Year"]==2020]

In [12]:
print(f"La media es {df_extration_filas["Annual change in population"].mean()}")

La media es 2124908.84765625


In [13]:
print(f"La mediana es {df_extration_filas["Annual change in population"].median()}")

La mediana es 34293.5


In [14]:
print(f"La varianza es {df_extration_filas["Annual change in population"].var()}")

La varianza es 90687335031103.64


In [15]:
print(f"La desviación estándar es {df_extration_filas["Annual change in population"].std()}")

La desviación estándar es 9522989.815761836


In [16]:
print(f"El mínimo es {df_extration_filas["Annual change in population"].min()}")

El mínimo es -494016.0


In [17]:
print(f"El máximo es {df_extration_filas["Annual change in population"].max()}")

El máximo es 75707590.0


In [18]:
df_extration_filas[df_extration_filas["Annual change in population"] == -494016.0]

,Entity,Code,Year,Annual change in population,Annual population change (Projected)
37269,Venezuela,VEN,2020,-494016.0,NaN


1.	**¿Cuál es el valor de la media?**
La media es 2124908.84765625
2.	**¿Cuál es el valor de la mediana?**
La mediana es 34293.5
3.	**¿Cuál es el valor de la desviación estándar?**
La desviación estándar es 9522989.815761836
4.	**¿Qué entidad presenta el valor mínimo y cuál el máximo?**
Venezuela presenta el mínimo y el máximo es 75707590.0

## Actividad 4 - Comprobación del entorno

In [19]:
import sys

print("Python:", sys.version)
print("Pandas:", pd.__version__)
print("NumPy:", np.__version__)


Python: 3.13.3 (tags/v3.13.3:6280bb5, Apr  8 2025, 14:47:33) [MSC v.1943 64 bit (AMD64)]
Pandas: 3.0.5
NumPy: 2.5.2


In [20]:
df[df['Entity'] == 'Mexico']

,Entity,Code,Year,Annual change in population,Annual population change (Projected)
22200,Mexico,MEX,1951,732546.0,NaN
22201,Mexico,MEX,1952,769686.0,NaN
22202,Mexico,MEX,1953,809069.0,NaN
22203,Mexico,MEX,1954,853770.0,NaN
22204,Mexico,MEX,1955,897588.0,NaN
...,...,...,...,...,...
22345,Mexico,MEX,2096,NaN,-743957.0
22346,Mexico,MEX,2097,NaN,-748746.0
22347,Mexico,MEX,2098,NaN,-752676.0
22348,Mexico,MEX,2099,NaN,-755662.0


In [21]:
df_extration_filas = df[df["Year"]==2020]
df_extration_filas

,Entity,Code,Year,Annual change in population,Annual population change (Projected)
69,Afghanistan,AFG,2020,1212852.0,NaN
219,Africa (UN),UN_AFR,2020,32815675.0,NaN
369,Albania,ALB,2020,-13061.0,NaN
519,Algeria,DZA,2020,747543.0,NaN
669,American Samoa,ASM,2020,-451.0,NaN
...,...,...,...,...,...
37719,Western Sahara,ESH,2020,10924.0,NaN
37869,World,OWID_WRL,2020,75707590.0,NaN
38019,Yemen,YEM,2020,1023451.0,NaN
38169,Zambia,ZMB,2020,545555.0,NaN


In [22]:
df_2006_2020_mex = df[(df["Entity"] == "Mexico") & (df["Year"]>=2006) & (df["Year"]<=2020)]
df_extraction_range = df_2006_2020_mex[["Entity", "Year", "Annual change in population"]]
df_extraction_range

,Entity,Year,Annual change in population
22255,Mexico,2006,1442162.0
22256,Mexico,2007,1520690.0
22257,Mexico,2008,1599930.0
22258,Mexico,2009,1625434.0
22259,Mexico,2010,1624178.0
22260,Mexico,2011,1619605.0
22261,Mexico,2012,1574705.0
22262,Mexico,2013,1525364.0
22263,Mexico,2014,1440693.0
22264,Mexico,2015,1288045.0


In [30]:
df_new = df[["Entity"]]
df_new

,Entity
0,Afghanistan
1,Afghanistan
2,Afghanistan
3,Afghanistan
4,Afghanistan
...,...
38395,Zimbabwe
38396,Zimbabwe
38397,Zimbabwe
38398,Zimbabwe


In [46]:
df.iloc[30:35]

,Entity,Code,Year,Annual change in population,Annual population change (Projected)
30,Afghanistan,AFG,1981,-1231727.0,NaN
31,Afghanistan,AFG,1982,-946206.0,NaN
32,Afghanistan,AFG,1983,-73395.0,NaN
33,Afghanistan,AFG,1984,272237.0,NaN
34,Afghanistan,AFG,1985,236633.0,NaN


In [47]:
df.loc[10000]

Entity                                  Equatorial Guinea
Code                                                  GNQ
Year                                                 2051
Annual change in population                           NaN
Annual population change (Projected)              46407.0
Name: 10000, dtype: object

In [48]:
df.loc[:,["Entity","Year","Annual change in population"]]

,Entity,Year,Annual change in population
0,Afghanistan,1951,103163.0
1,Afghanistan,1952,108441.0
2,Afghanistan,1953,108919.0
3,Afghanistan,1954,111251.0
4,Afghanistan,1955,119027.0
...,...,...,...
38395,Zimbabwe,2096,NaN
38396,Zimbabwe,2097,NaN
38397,Zimbabwe,2098,NaN
38398,Zimbabwe,2099,NaN


In [53]:
df.loc[(df["Entity"]=="Mexico") & (df["Year"]==2020)]

,Entity,Code,Year,Annual change in population,Annual population change (Projected)
22269,Mexico,MEX,2020,1036075.0,NaN


In [57]:
df.loc[0:5]

,Entity,Code,Year,Annual change in population,Annual population change (Projected)
0,Afghanistan,AFG,1951,103163.0,NaN
1,Afghanistan,AFG,1952,108441.0,NaN
2,Afghanistan,AFG,1953,108919.0,NaN
3,Afghanistan,AFG,1954,111251.0,NaN
4,Afghanistan,AFG,1955,119027.0,NaN
5,Afghanistan,AFG,1956,127322.0,NaN


In [64]:
df.iloc[:, 0:3]

,Entity,Code,Year
0,Afghanistan,AFG,1951
1,Afghanistan,AFG,1952
2,Afghanistan,AFG,1953
3,Afghanistan,AFG,1954
4,Afghanistan,AFG,1955
...,...,...,...
38395,Zimbabwe,ZWE,2096
38396,Zimbabwe,ZWE,2097
38397,Zimbabwe,ZWE,2098
38398,Zimbabwe,ZWE,2099


In [84]:
ds_apellido = df.loc[(df["Entity"]=="Morocco") & (df["Year"]>=2006) & (df["Year"]<=2023)]
ds_apellido

,Entity,Code,Year,Annual change in population,Annual population change (Projected)
23455,Morocco,MAR,2006,413034.0,NaN
23456,Morocco,MAR,2007,415289.0,NaN
23457,Morocco,MAR,2008,420301.0,NaN
23458,Morocco,MAR,2009,424010.0,NaN
23459,Morocco,MAR,2010,436230.0,NaN
23460,Morocco,MAR,2011,445576.0,NaN
23461,Morocco,MAR,2012,442661.0,NaN
23462,Morocco,MAR,2013,432327.0,NaN
23463,Morocco,MAR,2014,417211.0,NaN
23464,Morocco,MAR,2015,402806.0,NaN


In [91]:
print(f"Media:  {ds_apellido["Annual change in population"].mean()}")
print(f"Mediana:  {ds_apellido["Annual change in population"].median()}")
print(f"Desviación estándar:  {ds_apellido["Annual change in population"].std()}")
print(f"Mínimo:  {ds_apellido["Annual change in population"].min()}")
print(f"Máximo:  {ds_apellido["Annual change in population"].max()}")

Media:  408576.1666666667
Mediana:  415579.0
Desviación estándar:  25203.01788943725
Mínimo:  370235.0
Máximo:  445576.0


Bye